src/riak_kv_replrtq_snk.erl

```diff
@@ -236,50 +239,30 @@ set_workercount(QueueName, WorkerCount) ->
 %% @doc
 %% Change the number of concurrent workers whilst limiting the number of
 %% workers per peer
--spec set_workercount(queue_name(), pos_integer(), pos_integer())
-                                                            -> ok|not_found.
-set_workercount(QueueName, WorkerCount, PerPeerLimit)
-                                            when PerPeerLimit =< WorkerCount ->
-    gen_server:call(?MODULE,
-                    {worker_count, QueueName, WorkerCount, PerPeerLimit}).
+-spec set_workercount(
+    queue_name(), pos_integer(), pos_integer()) -> ok|not_found.
+set_workercount(
+        QueueName, WorkerCount, PerPeerLimit)
+        when PerPeerLimit =< WorkerCount ->
+    gen_server:call(
+        ?MODULE,
+        {worker_count, QueueName, WorkerCount, PerPeerLimit},
+        infinity
+    ).
 
 %%%============================================================================
 %%% gen_server callbacks
 %%%============================================================================
 
 init([]) ->
-    SinkEnabled =
-        app_helper:get_env(riak_kv, replrtq_enablesink, false),
+    SinkEnabled = app_helper:get_env(riak_kv, replrtq_enablesink, false),
     case SinkEnabled of
         true ->
-            SinkPeers =
-                app_helper:get_env(riak_kv, replrtq_sinkpeers, ""),
-            DefaultQueue =
-                app_helper:get_env(riak_kv, replrtq_sinkqueue),
-            SnkQueuePeerInfo = tokenise_peers(DefaultQueue, SinkPeers),
-            {SnkWorkerCount, PerPeerLimit} = get_worker_counts(),
-            Iteration = 1,
-            MapPeerInfoFun =
-                fun({SnkQueueName, SnkPeerInfo}) ->
-                    {SnkQueueLength, SnkWorkQueue} =
-                        determine_workitems(SnkQueueName,
-                                            Iteration,
-                                            SnkPeerInfo,
-                                            SnkWorkerCount,
-                                            min(SnkWorkerCount, PerPeerLimit)),
-                    SnkW =
-                        #sink_work{queue_name = SnkQueueName,
-                                    work_queue = SnkWorkQueue,
-                                    minimum_queue_length = SnkQueueLength,
-                                    peer_list = SnkPeerInfo,
-                                    max_worker_count = SnkWorkerCount},
-                    {SnkQueueName, Iteration, SnkW}
-                end,
-            Work = lists:map(MapPeerInfoFun, SnkQueuePeerInfo),
-            {ok, #state{enabled = true, work = Work}, ?INACTIVITY_TIMEOUT_MS};
+            gen_server:cast(?MODULE, initialise_work);
```

Comment:
This is not the intended API for a gen_server.
I guess you want to continue with initial_work as soon as init is done.
If that is the case, rather have init return 
```
{ok, #state{), {continue, initialise_work}}
```

Then you write `handle_continue` function
```
handle_continue(initialise_work, State) ->
  %% the code of the handle cast for initialise work here
```

Reply:
(Works in OTP 22 and above, checked that)

Reply:
Modified